app.js

```diff
@@ -90,6 +106,17 @@ app.get('/components*', function (req, res) {
   } else {
     // Show the isolated component preview
     path = path.replace(/preview/g, '')
+    if (path.indexOf('--') > -1) {
+      let noLeadingSlash = path.split(/^\/(.+)/)[1]
```

Comment:
This whole block is really hard to follow. Can it be refactored, split up or at least commented?